platform/src/main/java/org/hillview/storage/CassandraSSTableLoader.java

```diff
@@ -115,52 +124,61 @@ public CFMetaData getSSTableMetadata(Descriptor desc) throws Exception{
     /** Converting Cassandra's data type to Hillview data type */
     private static ColumnDescription getDescription(ColumnDefinition colDef) {
         String name = colDef.toString();
-        String colType = colDef.type.asCQL3Type().toString();
         ContentsKind kind;
-        switch (colType) {
-            case "boolean":
-            case "ascii":
-            case "inet":
-            case "text":
-            case "timeuuid":
-            case "uuid":
+        CQL3Type type = colDef.type.asCQL3Type();
+        assertTrue(type instanceof CQL3Type.Native);
```

Comment:
instanceof may be expensive, perhaps it's better not to test this, the next cast will throw if that does not work.